src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -372,14 +397,17 @@ private boolean canReadFrom(Job<?, ?> job, AbstractBuild<?, ?> build) {
     }
 
     // retrieve the "folder" (jenkins root if no folder used) for this build
-    private ItemGroup getItemGroup(AbstractBuild<?, ?> build) {
-        ItemGroup group = build.getProject().getRootProject().getParent();
-        return group;
-
+    private ItemGroup getItemGroup(Run<?, ?> build) {
+        if (build instanceof AbstractBuild) {
+            return ((AbstractBuild) build).getProject().getRootProject().getParent();
+        } else {
+            // TODO: hmmm ???
+            return build.getParent().getParent();
```

Comment:
This is fine. There is no equivalent of `getRootProject` for non-`AbstractProject`s, and none is needed. (The original code here is just the usual hacks to make things work with `matrix-project`.)